go/runtime/host/sgx/epid.go

```diff
@@ -52,17 +44,54 @@ func (ep *teeStateEPID) Update(ctx context.Context, sp *sgxProvisioner, conn pro
 	}
 	supportsAttestationV1 := (regParams.TEEFeatures != nil && regParams.TEEFeatures.SGX.PCS)
 
+	// TODO: Could consider implementing a strategy for more optimized endpoint selection,
+	// but (re-)attestation are not so frequent that it would matter much.
+	for i, iasClient := range sp.ias {
+		resp, err := ep.update(ctx, sp, conn, report, nonce, supportsAttestationV1, iasClient)
+		switch {
+		case err == nil:
+			return resp, nil
+		case i == len(sp.ias)-1:
+			return nil, err
+		default:
+			sp.logger.Warn("error obtaining attestation, trying next IAS server", "err", err)
```

Comment:
I prefer printing this warn for all errors.
```
		if err != nil {
			sp.logger.Warn("error obtaining attestation, trying next IAS server", "err", err)

			if i == len(sp.ias)-1 {
				return nil, err
			}

			select {
			case <-time.After(50 * time.Millisecond):
				continue
			case <-ctx.Done():
				return nil, ctx.Err()
			}
		}
```

Reply:
We should update metrics here (see `teeAttestationsFailed`).

Reply:
Should we? It depends on what the intended measurement of the metric is. Do we want to record a failed attestation if the problem is on (one of the) proxies?

Reply:
Maybe not. Actually, we can even add another metrics latter if we need to. Just that it is clear to operators what metrics mean.